contracts/protocore/Protocore.sol

```diff
@@ -114,4 +158,42 @@ contract Protocore is MosaicVersion, ValidatorSet, CoconsensusModule {
             openKernelHash
         );
     }
+
+
+    /* Private Functions */
+
+    /**
+     * @notice It calculates vote message hash.
+     */
+    function hashVoteMessage(
+        bytes32 _transitionHash,
+        bytes32 _sourceBlockHash,
+        bytes32 _targetBlockHash,
+        uint256 _sourceBlockNumber,
+        uint256 _targetBlockNumber
+    )
+        private
```

Comment:
Maybe this should be internal. `OriginProtocore` and `SelfProtocore` will call this function to generate the voteMessageHash for genesis.

Reply:
As mentioned in the ticket #183 (comment), It will only be used in `Protocore::_proposeLink`. So, I think we must keep it private.